mirrord/config/src/agent.rs

```diff
@@ -337,6 +350,14 @@ impl AgentImageFileConfig {
     }
 }
 
+/// <!--${internal}-->
+/// Specifies a secret reference for the agent pod.
+#[derive(Clone, Debug, PartialEq, Eq, JsonSchema, Deserialize, Serialize)]
+pub struct AgentPullSecret {
+    /// Name of the secret.
+    pub name: String,
+}
```

Comment:
```suggestion
///
/// Name of the secret.
pub struct AgentPullSecret(String);
```

The most complex suggestion to date.

Reply:
Yeah, it would be simpler. But would break compatibility :sweat_smile: